src/silx/gui/plot/LegendItem.py

```diff
@@ -0,0 +1,156 @@
+import logging
+
+from .. import qt
+from ..widgets.LegendIconWidget import LegendIconWidget
+from . import items
+from .items.core import HighlightedMixIn
+from .PlotWidget import PlotWidget
+
+_logger = logging.getLogger(__name__)
+
+
+class LegendItemWidget(qt.QWidget):
+    def __init__(self, parent, item: items):
+        super().__init__(parent)
+        self._item = item
+        self.setLayout(qt.QHBoxLayout())
+        self.setSizePolicy(
+            qt.QSizePolicy.Policy.Preferred, qt.QSizePolicy.Policy.Minimum
+        )
+        self.setMinimumWidth(150)
+        self.setMaximumWidth(300)
+        self.layout().setSpacing(20)
+        self.layout().setContentsMargins(10, 0, 10, 0)
+        item.sigItemChanged.connect(self._itemChanged)
+
+        self._icon = LegendIconWidget(parent=self)
+        self.layout().addWidget(self._icon)
+
+        self._label = qt.QLabel(self._item.getName())
+        self._label.setAlignment(qt.Qt.AlignLeft | qt.Qt.AlignVCenter)
+        self.layout().addWidget(self._label)
+        self.layout().addStretch()
+
+        self._label.setToolTip("Click to toggle visibility")
+        self._label.mousePressEvent = self._onLabelClicked
+        self._label.setCursor(qt.Qt.PointingHandCursor)
```

Comment:
You can set it for the whole widget
```suggestion
        self.setCursor(qt.Qt.PointingHandCursor)
```